Dockerfile

```diff
@@ -109,113 +93,140 @@ ARG DEBIAN_FRONTEND
+				apt-get update && apt-get install -y --no-install-recommends \
+				crossbuild-essential-armhf \
+				crossbuild-essential-arm64 \
+				crossbuild-essential-armel \
+			fi
 
 FROM cross-${CROSS} as dev-base
 
 FROM dev-base AS runtime-dev-cross-false
 ARG DEBIAN_FRONTEND
-RUN apt-get update && apt-get install -y --no-install-recommends \
-	libapparmor-dev \
-	libseccomp-dev \
-	&& rm -rf /var/lib/apt/lists/*
+RUN --mount=type=cache,sharing=locked,id=moby-cross-false-aptlib,target=/var/lib/apt \
+		--mount=type=cache,sharing=locked,id=moby-cross-false-aptcache,target=/var/cache/apt \
+			apt-get update && apt-get install -y --no-install-recommends \
+			libapparmor-dev \
+			libseccomp-dev
+
 FROM cross-true AS runtime-dev-cross-true
 ARG DEBIAN_FRONTEND
 # These crossbuild packages rely on gcc-<arch>, but this doesn't want to install
 # on non-amd64 systems.
 # Additionally, the crossbuild-amd64 is currently only on debian:buster, so
 # other architectures cannnot crossbuild amd64.
-RUN if [ "$(go env GOHOSTARCH)" = "amd64" ]; then \
-	apt-get update && apt-get install -y --no-install-recommends \
-		libseccomp-dev:armhf \
-		libseccomp-dev:arm64 \
-		libseccomp-dev:armel \
-		libapparmor-dev:armhf \
-		libapparmor-dev:arm64 \
-		libapparmor-dev:armel \
-		# install this arches seccomp here due to compat issues with the v0 builder
-		# This is as opposed to inheriting from runtime-dev-cross-false
-		libapparmor-dev \
-		libseccomp-dev \
-		&& rm -rf /var/lib/apt/lists/*; \
-	fi
+RUN --mount=type=cache,sharing=locked,id=moby-cross-true-aptlib,target=/var/lib/apt \
+		--mount=type=cache,sharing=locked,id=moby-cross-true-aptcache,target=/var/cache/apt \
+			if [ "$(go env GOHOSTARCH)" = "amd64" ]; then \
+				apt-get update && apt-get install -y --no-install-recommends \
+					libseccomp-dev:armhf \
+					libseccomp-dev:arm64 \
+					libseccomp-dev:armel \
+					libapparmor-dev:armhf \
+					libapparmor-dev:arm64 \
+					libapparmor-dev:armel \
+					# install this arches seccomp here due to compat issues with the v0 builder
+					# This is as opposed to inheriting from runtime-dev-cross-false
+					libapparmor-dev \
+					libseccomp-dev \
+			fi
+
 
 FROM runtime-dev-cross-${CROSS} AS runtime-dev
 
 FROM base AS tomlv
 ENV INSTALL_BINARY_NAME=tomlv
 COPY hack/dockerfile/install/install.sh ./install.sh
 COPY hack/dockerfile/install/$INSTALL_BINARY_NAME.installer ./
-RUN PREFIX=/build ./install.sh $INSTALL_BINARY_NAME
+RUN --mount=type=cache,target=/root/.cache/go-build \
+		--mount=type=cache,target=/go/pkg/mod \
+			PREFIX=/build ./install.sh $INSTALL_BINARY_NAME
 
 FROM base AS vndr
 ENV INSTALL_BINARY_NAME=vndr
 COPY hack/dockerfile/install/install.sh ./install.sh
 COPY hack/dockerfile/install/$INSTALL_BINARY_NAME.installer ./
-RUN PREFIX=/build ./install.sh $INSTALL_BINARY_NAME
+RUN --mount=type=cache,target=/root/.cache/go-build \
+		--mount=type=cache,target=/go/pkg/mod \
+			PREFIX=/build ./install.sh $INSTALL_BINARY_NAME
```

Comment:
Perhaps something to be done as part of this PR; I noticed earlier today that we have this `install.sh` script, but now that we have multi-stage builds (and in this PR, the opportunity to use `--mount=type=cache`), I think we can get rid of that. I _think_ the only reason we have that script is to cleanup the source-code we pulled from GitHub so that it doesn't end up in the layer; using a `cache` mount would prevent that, and (even better), would keep the cache around in case we have to build these stages again; https://github.com/moby/moby/blob/37498f009d8bf25fbb6199e8ccd34bed84f2874b/hack/dockerfile/install/install.sh#L6-L30

Reply:
Maybe separately.
I can see this being tricky because if the cache is populated with the git data then it's not a simple git clone to fetch things.